shower.js

```diff
@@ -14,12 +14,63 @@ window.shower = (function(window, document, undefined) {
 		slides = [],
 		progress = [],
 		timer,
-		isHistoryApiSupported = ! ! (window.history && window.history.pushState);
+		Modes = {
+			FULL : 'full',
+			LIST : 'list'
+		};
 
 	// Shower debug mode env, could be overridden before shower.init
 	shower.debugMode = false;
 
 	/**
+	 * Configuration store.
+	 *
+	 * @param {Object} [initialState]
+	 */
+	function State(initialState) {
+		initialState = initialState || {};
+		this._localStorageKey = 'shower';
+		this._store = this._loadFromLocalStorage() || initialState;
+	}
+
+	State.prototype = {
+		set : function(key, value) {
+			this._store[key] = value;
+			this._saveToLocalStorage();
```

Comment:
@zloylos, why do we need localStorage at all? Can't we just use `this._store`? As a user I find it a bit confusing that the mode is saved after page reload.